repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java

```diff
@@ -217,13 +212,6 @@ public Collection<GraphHelper.VertexInfo> getOwnedVertices(AtlasVertex entityVer
 
         while (vertices.size() > 0) {
             AtlasVertex        vertex = vertices.pop();
-            AtlasEntity.Status state  = getState(vertex);
-
-            //In case of purge If the reference vertex is active then skip it or else
-            //If the vertex marked for deletion, skip it
-            if (state == (isPurgeRequested ? ACTIVE : DELETED)) {
```

Comment:
In case deleteType:SOFT / HARD and state is DELETED, we would want "continue;" here right?

Reply:
Actually this was a redundant check. already we are checking this in skipVertexForDelete check and this function is only used in deleteEntities function. So I removed this from here.

Reply:
This check is not only being used for the vertex that we pass to getOwnedVertices flow but is used for other related vertices as well. Did you verify that?

Reply:
Yeah right, we should add this check. So I will skip this if and only if entity status is ACTIVE and we are not purging. Updated one more check for edge deletion also. The check is similar. Tested this.